### ISSUE TITLE
engine.release(); 异常

### ISSUE THREAD

**Issue by yirenkeji555** [AgoraIO-Extensions/react-native-agora#538]
<img width="385" alt="image" src="https://user-images.githubusercontent.com/48214423/195060575-0edee72f-ac92-4c4a-a2c7-65452e74d218.png">

example 中
hook/JoinChannelVideo/JoinChannelVideo.tsx 使用报错

**Comment by yirenkeji555** [comment]
<img width="965" alt="image" src="https://user-images.githubusercontent.com/48214423/195061775-b415570a-2ec2-4007-946f-7764013c54c0.png">

**Comment by EkaanshArora** [comment]
@LichKing-2234 I can reproduce the same error with:
```tsx
      try {
        console.log('release type: ', typeof engine.current.release);
        engine.current.release();
      } catch (e) {
        console.log('release error: ', e);
      }
```
log:
```bash
 LOG  release type: function
 LOG  release error: [TypeError: undefined is not a function]
 ```

**Comment by LichKing-2234** [comment]
@EkaanshArora What is `engine.current`?

**Comment by radko93** [comment]
@LichKing-2234 it's likey a `useRef` usage

**Comment by yirenkeji555** [comment]
@EkaanshArora 
but in demo file use "useState"
const [engine] = useState(createAgoraRtcEngine());

**Comment by EkaanshArora** [comment]
For immutable values in React it's recommended to use a ref. Nevertheless, the error with the release method is consistent across both usages.

**Comment by yirenkeji555** [comment]
@EkaanshArora 
I already use "useRef", but not work.
<img width="1530" alt="image" src="https://user-images.githubusercontent.com/48214423/195645292-6d3ab248-2c21-4fe4-9ef3-c773860b893d.png">

<img width="1334" alt="image" src="https://user-images.githubusercontent.com/48214423/195645046-bb1e80a4-5217-4338-9c5a-f08577e4b567.png">

**Comment by EkaanshArora** [comment]
@LichKing-2234 I updated our v3.x [quickstart example](https://github.com/AgoraIO-Community/Agora-RN-Quickstart/tree/4.x) to v4.0.0 it reproduces this issue. You can test it [here](https://github.com/AgoraIO-Community/Agora-RN-Quickstart/blob/0d98bbbbd196bd0e69434d9ca444d638c4d7917c/App.tsx#L89).

```
 LOG  JoinChannelSuccess {"channelId": "test", "localUid": 4131065458}
 LOG  UserJoined {"channelId": "test", "localUid": 4131065458} 2620116037
 LOG  release error: [TypeError: undefined is not a function]
```

**Comment by LichKing-2234** [comment]
Will be fixed in the 4.1.0 version.

**Comment by vietdung123** [comment]
> Will be fixed in the 4.1.0 version.

I have same issues. How to most succinctly unregister all listenners when engine.release() throw above error? @LichKing-2234

**Comment by LichKing-2234** [comment]
@vietdung123 you can call `removeAllListeners` before `release`
